packages/teleterm/src/services/pty/ptyService.ts

```diff
@@ -55,18 +56,27 @@ export default function createPtyService(
   };
 }
 
-function buildOptions(settings: RuntimeSettings, cmd: PtyCommand): PtyOptions {
+async function buildOptions(
+  settings: RuntimeSettings,
+  cmd: PtyCommand
+): Promise<PtyOptions> {
   const env = {
+    ...process.env,
     TELEPORT_HOME: settings.tshd.homeDir,
   };
 
+  const localShellEnv = {
+    ...env,
+    ...(await resolveShellEnvCached(settings.defaultShell)),
+  };
```

Comment:
I'm not sure if this should be this way but I'm running out of time for today, I'll check it tomorrow morning + I have a solution for setting the correct cwd instead of opening the shell in `/`.

Reply:
Okay, so I think the priority should be this:

1. TELEPORT_HOME
2. resolveShellEnvCached
3. process.env

What was clear to you but took a second for me is that obviously we want `resolveShellEnvCached` to override whatever path is set in `process.env`, because that's the problem we're solving here – launching the app from outside of the terminal doesn't set path correctly.

Now, I wonder what we should do about `TELEPORT_HOME`. The current code is going to override it with whatever comes from user's env. Regular Teleport users probably won't have this env var set anyway, but powerusers might in which case they might not expect Teleterm to use it and it'll mess with their tsh directory. OTOH if always override `TELEPORT_HOME` with `settings.tshd.homeDir`, users won't be able to easily change that. But for the preview release that should be okay? If someone really needs that they can probably symlink `~/Library/Application\ Support/teleterm/tsh` to whatever they want. 🤔

Now that I think about it, we probably don't even need to include stuff from `process.env` when launching a shell. `resolveShellEnvCached` already gives us all the env vars we need. That being said, it's hard for me to say if this is going to have any unforeseen consequences, so I cannot really recommend that we do it.

Reply:
I don't have any strong opinion on envs priority, but it seems ok to override `TELEPORT_HOME` with `settings.tshd.homeDir`.
Regarding `process.env` from what I see all of its properties are overridden by the resolved shell, but I'm not sure if we should remove it, I think I'd leave it as it is.